Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -329,7 +329,8 @@ double vtkMRMLMarkupsCurveNode::GetCurveLength(vtkPoints* curvePoints, bool clos
     previousPoint[2] = nextPoint[2];
     }
   // Add length of closing segment
-  if (closedCurve && (numberOfCurvePoints < 0 || numberOfCurvePoints >= curvePoints->GetNumberOfPoints()))
+  if (closedCurve && startCurvePointIndex <= lastCurvePointIndex &&
```

Comment:
This entire logic is becoming just too hard to read (and therefore error-prone and may be misinterpreted by users when they try to understand what the code does). What do you think about setting an `addClosingSegment` Boolean flag during the computation of lastCurvePointIndex? For example, the flag could be enabled by default if curve is closed but it would be set to false if it is detected that only a partial curve length is requested that does not include the closing segment (e.g., it could be set around line 316 where we update the `lastCurvePointIndex`).

In the end then we would just check the `addClosingSegment` flag and add the closing segment length.

Reply:
Sounds good.  I just force pushed.